develop/s3_handler.py

```diff
@@ -15,23 +18,32 @@
 
 
 class S3Handler:
-    def __init__(self, s3_client, bucket_name=BUCKET_NAME):
+    def __init__(self, s3_client, bucket_name=BUCKET_NAME, divider_column=None):
         self.s3_handler = s3_client
         self.bucket_name = bucket_name
-
-        # Configure the logging
+        self.divider_column = divider_column
 
     def upload(self, key, data):
         try:
-            self.s3_handler.put_object(
-                Bucket=self.bucket_name,
-                Key=key,
-                Body=data
-            )
+            # If there's no divider, upload it as one file into the bucket
+            if self.divider_column is None:
+                self.s3_handler.put_object(Bucket=self.bucket_name, Key=key, Body=data.getvalue())
+            else:
+                partitions = self._divide_data_into_partitions(data)
+                for partition, partition_data in partitions.items():
+                    self.s3_handler.put_object(Bucket=self.bucket_name, Key=f"{partition}/{key}",
+                                               Body=partition_data.getvalue())
             logger.info(f"Uploaded to S3: s3://{self.bucket_name}/{key}")
         except NoCredentialsError:
             logger.error("Failed to write to S3. Check your AWS credentials.")
 
-    def upload_file(self, file_path, file_name):
-        with open(file_path, "rb") as f:
-            self.s3_handler.upload_fileobj(f, self.bucket_name, file_name)
+    def _divide_data_into_partitions(self, data):
```

Comment:
maybe use the groupby function in the dataframe class?
example - https://chat.openai.com/share/776c0d11-1a6c-4005-8ff7-9ea63f765b74